Added derivatives of Matern-5/2.

opkoisti opened the pull request.

Commit:
Added derivatives of Matern-5/2.

jpvanhat reviewed (commented):
There seems to be something wrong with the derivative observations when using selectedVariables and additive model. I run a test with demo_derivatives.m (can be found from develop-branch) so that I changed all gpcf_sexp to gpcf_matern52. Everything worked fine until line 474 where gradient check returned

>> gradcheck(gp_pak(gp), @gp_e, @gp_g, gp, xt, yt);
Checking gradient ...

   analytic   diffs     delta

   -4.3433   -4.3433    0.0000
  -10.5201  -12.3472    1.8271
   -5.8705   -5.8705    0.0000
   -8.8214   -7.9206   -0.9008
   -1.3060   -1.3060    0.0000

Further down on line 577 gradient check fails as well:

>> gradcheck(gp_pak(gp), @gp_e, @gp_g, gp, xt, yt, 'z', z);
Checking gradient ...

   analytic   diffs     delta

   -2.3302   -2.3299   -0.0002
   -6.6229   -6.0056   -0.6173
   -1.6128   -1.6128   -0.0000
   -1.6812    0.4867   -2.1679
   -0.7463   -0.7463    0.0000

Could you check these OP

Commit:
Added "selectedVariables" in subfunction "ginput" in Matern-3/2

Commit:
Added "selectedVariables" in subfunction "ginput" in Matern-5/2

Also corrected a mistake on row 409: 'selectedVariable' -> 'selectedVariables'.

opkoisti commented:
I could not reproduce these errors for some reason. However, I corrected one mistake on line 409 in subfunction "trvar", where "selectedVariables" is skipped because of a typo. I hope this solves the problem you found.

In addition, I added "selectedVariables" functionality also in subfunction "ginput". It was missing also in Matern-3/2 already before adding the derivatives, so I suspect that this might not have affected on your tests.

jpvanhat commented:
Everything seems to work now. I will approve the pull request.
